atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -30,43 +28,29 @@ def requires_asap_wrapper(request, *args, **kwargs):
             # that is the case to properly support Python 3.
             if isinstance(auth_header, six.string_types):
                 auth_header = auth_header.encode(encoding='iso-8859-1')
-            auth = auth_header.split(b' ')
-            if not auth or len(auth) != 2:
+            try:
+                scheme, auth = auth_header.split(b' ')
+            except ValueError:
+                scheme = b''
+
+            if scheme.lower() != b'bearer':
                 return HttpResponse('Unauthorized', status=401)
 
-            message = None
-            exception = None
             try:
-                asap_claims = parse_jwt(verifier, auth[1])
+                asap_claims = parse_jwt(verifier, auth)
                 verify_issuers(asap_claims, issuers)
                 request.asap_claims = asap_claims
-                return func(request, *args, **kwargs)
-            except RequestException as e:
-                # Error communicating to get key
-                message = 'Unauthorized: Communications error retrieving key'
-                exception = e
-            except PrivateKeyRetrieverException as e:
-                # Error parsing or getting private key
-                message = 'Unauthorized: Unable to retrieve private key'
-                exception = e
-            except PublicKeyRetrieverException as e:
-                # Error parsing or getting public key
-                message = 'Unauthorized: Unable to retrieve public key'
-                exception = e
-            except InvalidIssuerError as e:
-                message = 'Unauthorized: Invalid token issuer'
-                exception = e
+            except InvalidAudienceError as e:
+                # Audience mismatch
+                message = 'Forbidden: %s' % e
+                return HttpResponse(message, status=403)
             except InvalidTokenError as e:
-                # Something went wrong with decoding the JWT
-                message = 'Unauthorized: Invalid token'
-                exception = e
-            if message is not None:
-                logger = logging.getLogger(__name__)
-                logger.error(message,
-                             extra={'original_message': str(exception)})
-
+                # Invalid token
+                message = 'Unauthorized: %s' % e
                 return HttpResponse(message, status=401)
```

Comment:
Shouldn't we be including a WWW-authenticate response header as per [RFC-6750](https://tools.ietf.org/html/rfc6750#section-3), considering we are using Bearer tokens?